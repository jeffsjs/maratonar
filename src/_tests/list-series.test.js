import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ListSeries from '../components/list-series';

import configureStore from 'redux-mock-store';
const mockStore = configureStore();

const initialState = {	
		loading: false,
		error: false,
		errorMessage: '',
		series: [
			{
				id: '3',
				imdbId: 'tt0475784',
				title: 'Westworld',
				numberOfSeasons: 3,
				releaseDate: '2017-01-28'
			},
			{
				id: '5',
				imdbId: 'tt6468322',
				title: 'La Casa de Papel',
				numberOfSeasons: 3,
				releaseDate: '2017-08-28'
			},
			{
				id: '6',
				imdbId: 'tt0098936',
				title: 'Twin Peaks',
				numberOfSeasons: 2,
				releaseDate: '1991-08-03',
				img:
					'https://m.media-amazon.com/images/M/MV5BMTExNzk2NjcxNTNeQTJeQWpwZ15BbWU4MDcxOTczOTIx._V1_SX300.jpg'
			}
		],
		seasons: [],
		episodes: [],
		selectedSerie: '3',
		dateStart: '',
		dateNextEpisode: '',
		nextEpisode: [],
		marathonDuration: 0,
		daysToNextEpisode: 0,
		minutesToNextEpisode: 0,
		hoursToNextEpisode: 0

};

const store = mockStore(initialState);

describe('ListSeries component', () => {
	it('Render ListSeries', () => {
		const wrapper = shallow(<ListSeries store={store} />).dive();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
