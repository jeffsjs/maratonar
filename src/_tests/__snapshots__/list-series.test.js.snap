// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSeries component Render ListSeries 1`] = `
<withRouter(ListSeries)
  loadPosterSeason={[Function]}
  loadingSeries={[Function]}
  showSerie={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
